src/NSubstitute.Analyzers.Shared/CodeFixProviders/AbstractReceivedInReceivedInOrderCodeFixProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Operations;
+using Document = Microsoft.CodeAnalysis.Document;
+
+namespace NSubstitute.Analyzers.Shared.CodeFixProviders;
+
+internal abstract class AbstractReceivedInReceivedInOrderCodeFixProvider : CodeFixProvider
+{
+    public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DiagnosticIdentifiers.ReceivedUsedInReceivedInOrder);
+
+    public override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var diagnostic = context.Diagnostics.FirstOrDefault(diag => diag.Descriptor.Id == DiagnosticIdentifiers.ReceivedUsedInReceivedInOrder);
+        if (diagnostic != null)
+        {
+            var codeAction = CodeAction.Create("Remove redundant checks", ct => CreateChangedDocument(ct, context, diagnostic), nameof(AbstractReceivedInReceivedInOrderCodeFixProvider));
```

Comment:
@dtchepak should we include check name in title? E.g "Remove redundant Received check", "Remove redundant ReceivedWithAnyArgs check" etc

Reply:
I think "Remove redundant Received checks" would be great. (I don't think we need to change it for `ReceivedWithAnyArgs`, they are all `Received*` family checks)

Reply:
sounds good, changed